### DISCUSSION TITLE
Modal custom styles (and z-index)

### DISCUSSION THREAD

**Discussion by joasmi723** [cengage/react-magma#31] [Ideas]
The modal component consists of an outer `ModalContainer` element, and inside it has a `ModalBackdrop`, and a `ModalContent` element.  Right now, when a consumer passes styles into the `Modal` component, those styles get applied to the `ModalContent`.  There is no way to apply custom styles to the `ModalContainer` or the `ModalBackground`.

Both the `ModalContainer` and the `ModalBackground`  have `z-index` values hard coded on them.  This is causing issues in applications where other ui elements have z-indexes applied.  

If we allow custom styles to be applied to the container, backdrop and content individually, that will allow consumers to set a custom z-index.   

Would this be the best approach for solving the issue with the z-index and the issue of custom styles in general?  Are there any other approaches we might take?

**Comment by rcgillet** [comment]
I think this is just something we need to expand on not only for `Modal`, but for every composed component. As we move forward I think we should make sure consumers are able to customize each piece. That was we don't have people trying to do hacky workarounds. 

That being said, hopefully we can put a lot of things in the `theme` as well while changing this.
